docs/AVS-Guide.md

```diff
@@ -0,0 +1,85 @@
+
+Let us illustrate the usage of this facility with an example: A staker who has opted-in to an AVS via delegation, wants to partially withdraw some of its stake. Whenever the staker partially withdraw some of its stake, it affects all the AVSs uniformly that its operator is participating in. The series of steps for partially withdrawing stake is as follows:
+ - The staker queues their withdrawal request with EigenLayer. The staker can place this request by calling  `queueWithdrawal(..)` in the EigenLayer's `StrategyManager.sol`.
+ - The operator, noticing an upcoming change in their delegated stake, notifies the AVS about this change. To do this, the operator triggers the AVS to call the `recordStakeUpdate(..)` in the AVS's ServiceManager contract which in turn accesses `recordStakeUpdate(..)` in the EigenLayer's Slasher contract.  On successful execution of this call, the event `MiddlewareTimesAdded(..)` is emitted.
+- The AVS provider now is aware of the change in stake, and the staker is free to complete their withdrawal.
+
+The following figure illustrates the above flow: 
+![Stake update](./images/staker_withdrawing.png)
+
+### *Deregistering from AVS*
+In order for any EigenLayer operator to be able to de-register from a AVS, EigenLayer provides the interface `recordLastStakeUpdateAndRevokeSlashingAbility(..)`. Essentially, in order for an operator to deregister from a AVS, the operator has to call `recordLastStakeUpdateAndRevokeSlashingAbility(..)` in EigenLayer's `Slasher.sol` via the AVS's ServiceManager contract. It is important to note that the latest block number until which the operator is required to serve tasks for the service must be known by the service and included in the ServiceManager's call to `Slasher.recordLastStakeUpdateAndRevokeSlashingAbility`.
+
+The following figure illustrates the above flow in which the operator calls the `deregister(..)` function in a sample Registry contract.
+![Operator deregistering](./images/operator_deregister.png)
+
+### *Slashing*
+As mentioned above, EigenLayer is built to support slashing as a result of an on-chain-checkable, objectively attributable action. In order for a AVS to be able to slash an operator in an objective manner, the AVS needs to deploy a DisputeResolution contract which anyone can call to raise a challenge against an EigenLayer operator for its adversarial action. On successful challenge, the DisputeResolution contract calls the `ServiceManager.freezeOperator(..)`; the ServiceManager in turn calls `Slasher.freezeOperator(..)` to freeze the operator in EigenLayer. EigenLayer's Slasher contract emits a `OperatorFrozen(..)` event whenever an operator is (successfully) frozen
```

Comment:
"calls the `ServiceManager.freezeOperator(..)`" => "calls `ServiceManager.freezeOperator(..)`" (delete the "the").

any thoughts on the general pattern of writing functions like this? I feel like I made a general suggestion and explained how to apply it in a single place, but it only got applied in that single place -- it looks weird / stands out like this :/